experimental/zerotrie/src/reader.rs

```diff
@@ -410,25 +410,51 @@ pub(crate) fn get_parameterized<T: ZeroTrieWithOptions + ?Sized>(
 ///
 /// The input-output argument `trie` starts at the original trie and ends pointing to
 /// the sub-trie reachable by `c`.
-pub(crate) fn step_ascii_bsearch_only(trie: &mut &[u8], c: u8) {
+#[inline]
+pub(crate) fn step_parameterized<T: ZeroTrieWithOptions + ?Sized>(
+    trie: &mut &[u8],
+    c: u8,
+) -> Option<u8> {
+    // BinarySpans is tricky to implement because the state can no longer be simply a trie
+    debug_assert!(
```

Comment:
Should this return `None` here without assertions?

Reply:
Without assertions it returns None on line 468, if it actually hits a span node.

Reply:
There's already a debug assert there, why this as well? There should at least be a comment linking to the line and what the gigo value is.

Reply:
I expanded these comments